infrastructure/server-setup/playbook-3.yml

```diff
@@ -113,20 +115,32 @@
         group: 1000
         mode: 0400
 
-    - name: 'Check file for encrypted file system exists'
+    - name: 'Precheck if encrypted file system exists so we dont try to bootstrap'
       stat:
         path: /cryptfs_file_sparse.img
         get_checksum: False
         get_md5: False
-      register: st
+      register: encryptedFileSystemPreCheck
 
     - name: 'Bootstrap encrypted data folder'
-      script: ../cryptfs/bootstrap.sh -s 100g -p {{disk_encryption_key}}
-      when: (not st.stat.exists) and encrypt_data
+      script: ../cryptfs/bootstrap.sh -s {{encrypted_disk_size}} -p {{disk_encryption_key}}
+      when: (not encryptedFileSystemPreCheck.stat.exists)
+
+    - name: Wait for encrypted file system
+      ansible.builtin.wait_for:
+        path: /cryptfs_file_sparse.img
+        state: present
+
+    - name: 'Register encrypted file system'
+      stat:
+        path: /cryptfs_file_sparse.img
+        get_checksum: False
+        get_md5: False
+      register: encryptedFileSystemPostCheck
```

Comment:
Is this block and the following `when: encryptedFileSystemPostCheck.stat.exists` needed anymore after the wait for `/cryptfs_file_sparse.img` is over?